cmd/migrations.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+)
+
+var (
+	migrations = map[int]func() error{0: migrateTopLevelFiles}
+	showMsg    = true
+	migRunning = false
+)
+
+// poor-man's migrations: there are no rollbacks (for now)
+func runMigrations() error {
+	// by using an int index we can sort of "enforce" an order
+	// with just an array it could easily happen that someone
+	// prepends a new migration at the front instead of the bottom
+	for i := 0; i < len(migrations); i++ {
+		err := migrations[i]()
+		if err != nil {
+			return fmt.Errorf("migration #%d failed: %w", i, err)
+		}
+	}
+	if migRunning {
+		ux.Logger.PrintToUser("Update process successfully completed")
+		migRunning = false
+	}
+	return nil
+}
+
+// Up to version 1.0.0 the sidecar and genesis files were stored at
+// {baseDir} in the top-level.
+// Due to new requirements and evolution of the tool, we now store
+// every subnet-specific file in {baseDir}/subnets/{subnetName}
+func migrateTopLevelFiles() error {
```

Comment:
Can you write a test for this migration?

Reply:
I guess I should :smile: